### ISSUE TITLE
Can't Pass ROHF `ref_wfn` to MRCC

### ISSUE THREAD

**Issue by JonathonMisiewicz** [psi4/psi4#2308]
For some reason, the following script raises a segfault:

```
molecule {
O
H 1 1.0 
H 1 1.0 2 104.5
}

set {
reference rohf
basis cc-pVDZ
scf_type pk
freeze_core true
}

_, wfn = energy('bccd(t)',return_wfn = True)
energy('mrccsdt', ref_wfn=wfn)
```

It runs fine when I use RHF Brueckner or ROHF SCF, so it's the ROHF Brueckner combination that's guilty.

**Comment by JonathonMisiewicz** [comment]
A cursory run through the debugger indicates the problem isn't actually specific to the `cc` code. The problem is that `mrcc` [assumes](https://github.com/psi4/psi4/blob/da36e2119babb65d893bc89494ad1dd859f750dd/psi4/src/psi4/mrcc/mrcc.cc#L746-L749) that if the reference is set to ROHF, it is receiving an ROHF wavefunction that needs to be semicanonicalized. This is a bad assumption that needs to be removed from the code. Determining what to do instead is less obvious.

**Comment by JonathonMisiewicz** [comment]
Pinging @lothian, who said he'd look into the issue.

**Comment by JonathonMisiewicz** [comment]
Thanks for the update! It sounds to me like the `mrcc` module should move away from determining RHF/UHF/ROHF from the value of the global and read it off the wavefunction. The ROHF-B-CC wavefunction should have `same_a_b_orbs_ = False`.

**Comment by lothian** [comment]
Yes, I think that's right.  I think the right place for this is in `ccrtransort/cctransort.cc`, since that appears to be the first point in the B-CC workflow in which ROHF is effectively converted to UHF.  Would you agree?

**Comment by JonathonMisiewicz** [comment]
The reference wavefunction shouldn't be changed, but cloning the wavefunction _before_ `cctransort`, and then having `cctransort` set `same_a_b_orbs_` sounds reasonable to me. I'd also agree with the driver changing this right before or right after `cctransort`, depending on what is most convenient for the internal logic of `cctransort`.

`Wavefunction` doesn't define a way to set change `same_a_b_orbs_`, but we can add one easily.

**Comment by lothian** [comment]
I would be ok with either approach as well. How do you want to proceed, Jonathan?  Do you want to do this quickly, or do you want me to do this slowly? ;)

**Comment by JonathonMisiewicz** [comment]
I think the best course of action is for me to take this as part of my foray into the `cc` mega-module, after I'm through with the test pass.

Incidentally, the current plan for my `cc` module pass is: get `cc` tests off perl, this issue, move variables to `wfn`, move `ccdensity` to Matrix objects, then give `ccdensity` a Wavefunction. There's a lot more that could be done, but my primary interest is in syncing up densities across multiple modules.

**Comment by lothian** [comment]
Sounds like a great plan.  I'll try to be responsive when/if you have questions.
